### ISSUE TITLE
/socket.io/socket.io.js 404 - with zappa.app function, opions

### ISSUE THREAD

**Issue by eddyparkinson** [zappajs/zappajs#122]
/socket.io/socket.io.js  is not found when I use  
` zappa.app function, opions`
but is found with
`zappajs port, host, options, function`

Code for http2 server push
```

options.disable_io = false

if keyfile? and certfile?
  # http2 server push
  
  h2options =
    key: fs.readFileSync keyfile
    cert: fs.readFileSync certfile
  http2 = require \spdy
  http2
    .createServer(h2options, ((require \zappajs).app zappainitialize, options).app  )
    .listen port, host, ->
else 
  (require \zappajs) port, host, options, zappainitialize

```

I have tried setting options.disable_io =false and options.io = {}
but still get 404 fine not found.

**Comment by eddyparkinson** [comment]
This is the cause:
http://stackoverflow.com/questions/19426882/node-js-socket-io-socket-io-js-not-found

not sure of the best way to address this, is it better to fix in zappajs or for my code to load socket.io

**Comment by shimaore** [comment]
There's a chicken-and-egg problem with creating the SPDY server (which requires a handler), the Socket.io handler (which requires a server) and the Zappa server (which creates the server and the handler). The proper solution is to expose the `server` and `io` items from zappajs, I'll be posting this shortly with an example.

There's a [separate discussion](https://stackoverflow.com/questions/12103628/spdy-as-replacement-for-websockets#12103981) about using [sse](https://github.com/mosdefbiz/node-easysse#demo) rather than websocket for this purpose which might be of interest.

**Comment by shimaore** [comment]
@eddyparkinson have a look at https://github.com/zappajs/zappajs/blob/6.x/examples/spdy.coffee , which is probably the simplest way to implement this — the new options, especially `http_module`, are available in v6.1.0.

Alternatively, if you'd like to keep control of the server creation, you can give it to zappa as an option and bind the application to the server later:

```
server = http2.createServer(options)
zapp = zappa.app {server}, ->
  @get '/', -> @send 'hello'
server.on 'request', zapp
```
